File: Challenge/ANAGRAMA/anagrama.ts
const sonAnagramas = (palabra1: string, palabra2: string): boolean => {
  const palabra1Ordenada = palabra1.split("").sort().join("");
  const palabra2Ordenada = palabra2.split("").sort().join("");

  return palabra1Ordenada === palabra2Ordenada;
};

console.log(sonAnagramas("amor", "roma"));

File: Challenge/ANAGRAMA/anagrama.dart
void main(List<String> args) {
  print(anagrama('amor', 'roma'));
  print(anagrama('amor', 'romaa'));
}

bool anagrama(String a, String b) {
  a = a.toLowerCase();
  b = b.toLowerCase();

  if (a.length != b.length) return false;

  for (var i = 0; i < a.length; i++) {
    if (b.indexOf(a[i]) == -1) return false;
  }

  return true;
}
